### PR TITLE
Add support for `display_order_weight` to order pipelines within a gr…

### PR DESCRIPTION
**Pull request by ketan**
…oup.

### CODE REVIEW

File: dsl/src/main/java/cd/go/contrib/plugins/configrepo/groovy/dsl/Pipeline.java [hunk 1]
```diff
@@ -57,6 +58,14 @@
     @NotEmpty
     private String group;
 
+    /**
+     * Use this integer to order the display of pipelines on the dashboard. If multiple pipelines have the same
+     * {@code displayOrder} value, their order relative to each other will be indeterminate.
+     */
+    @JsonProperty("display_order_weight")
+    @PositiveOrZero
```
Comment (arvindsv) [inline dsl/src/main/java/cd/go/contrib/plugins/configrepo/groovy/dsl/Pipeline.java]: @ketan I think this annotation can be removed. This value can be negative too.

**Comment by arvindsv** [comment]
@ketan It also looks like the `displayOrder` is mandatory. Doesn't work when not found. I've mentioned the Groovy script I tried with, above. When I remove the `displayOrder` line for one pipeline, it fails:

```
test.gocd.groovy;
1. Unable to parse file test.gocd.groovy.   - pipelines[1].displayOrder must be greater than or equal to 0
  - pipelines.elements[1].displayOrder must be greater than or equal to 0
```

**Comment by ketan** [comment]
I think I've addressed your concerns. Do you mind taking a look once more?

**Comment by arvindsv** [comment]
@ketan I tried it out again. Everything looks good.

**Comment by ketan** [comment]
I've cut a release with the display order fix.

- Ketan

**Pull request merged**